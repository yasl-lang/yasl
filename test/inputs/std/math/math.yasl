##true\n180.0\n0.20788\n1.0\ntrue\n10.0\ntrue\ntrue\ntrue\n48\n15360\ntrue\nfalse\ntrue\n

fn inEpsilon(a, b) {
	return math.abs(a - b) < 10**-10
}

echo inEpsilon(0, 10**-20) && inEpsilon(0.1**20, 0)

echo math.deg(math.pi)

echo math.exp(-math.pi/2)

echo 2*math.sin(math.rad(45))**2

echo math.ceil(0.5) == 1 && math.ceil(0) == 0 && math.floor(0.5) == 0 && math.floor(0) == 0

echo math.log(1024)/math.log(2)

echo math.sqrt(0.5) == math.cos(math.pi / 4)

echo inEpsilon(math.asin(0.5), math.rad(30))

echo inEpsilon(math.tan(1984), math.sin(1984)/math.cos(1984))

## Number theory
echo math.gcd(3072, 240)

echo math.lcm(3072, 240)

# Type casts to integer
echo math.isprime(15485867.6)
echo math.isprime(15485869.1)

const x = 18408990
echo math.isprime(x-1) && math.isprime(x+1)

echo math.clamp(50.5, 0, 100);
echo math.clamp(20.1, 50, 100);
echo math.clamp(100.9, 0, 100.5);
echo math.clamp(20.8, 50.2, 100.9);
