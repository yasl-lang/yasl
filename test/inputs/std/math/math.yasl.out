true
180.0
0.20788
1.0
true
10.0
true
true
true
48
15360
true
false
true
50.5
50
100.5
50.2
